src/drawers/drawer.ts

```diff
@@ -76,7 +76,14 @@ export class Drawer {
    * @param{any[]} data The data to be drawn
    */
   private _bindSelectionData(data: any[]) {
-    let dataElements = this.selection().data(data);
+    // if the dataset has a key, use it when binding the data   
+    let dataElements: d3.selection.Update<any>;
+    if (this._dataset && this._dataset.key()) {
+      dataElements = this.selection().data(data, this._dataset.key());
+    } else {
+      dataElements = this.selection().data(data);
+    }
```

Comment:
I think I see what you are getting at.  I think the idea that you are proposing definitely makes sense.  I am mainly curious in what you thought about us removing that previous workflow inherently (even though it seems that if users were to use `noConstancy` then they would be in the clear in the first place).